monai/transforms/spatial/dictionary.py

```diff
@@ -1338,6 +1362,8 @@ def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.nda
         self.randomize()
         d = dict(data)
         if not self._do_transform:
+            for key in self.keys:
+                self.push_transform(d, key, extra_info={"zoom": self._zoom})
```

Comment:
perhaps extra_info `self._zoom` could have different sizes (on/off _do_transform) in a batch loader, then they are not compatible with the collate https://github.com/Project-MONAI/MONAI/issues/1798

Reply:
Yeah, you're right. You can see there's a few `ensure_tuple(self._zoom)` that come a few lines later, so they just need to be moved ahead.